src/credentials/providers/ssoCredentialProvider.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { Credentials, SSO } from 'aws-sdk'
+import { getLogger } from '../../shared/logger'
+import { SsoAccessTokenProvider } from '../sso/ssoAccessTokenProvider'
+
+export class SsoCredentialProvider {
+    private ssoAccount: string
+    private ssoRole: string
+    private ssoClient: SSO
+    private ssoAccessTokenProvider: SsoAccessTokenProvider
+
+    constructor(ssoAccount: string, ssoRole: string, ssoClient: SSO, ssoAccessTokenProvider: SsoAccessTokenProvider) {
+        this.ssoAccount = ssoAccount
+        this.ssoRole = ssoRole
+        this.ssoClient = ssoClient
+        this.ssoAccessTokenProvider = ssoAccessTokenProvider
+    }
+
+    public async refreshCredentials(): Promise<Credentials> {
+        try {
+            const accessToken = await this.ssoAccessTokenProvider.accessToken()
+            const roleCredentials = await this.ssoClient
+                .getRoleCredentials({
+                    accountId: this.ssoAccount,
+                    roleName: this.ssoRole,
+                    accessToken: accessToken.accessToken,
+                })
+                .promise()
+
+            return new Credentials({
+                accessKeyId: roleCredentials.roleCredentials?.accessKeyId!,
+                secretAccessKey: roleCredentials.roleCredentials?.secretAccessKey!,
+                sessionToken: roleCredentials.roleCredentials?.sessionToken,
+            })
+        } catch (err) {
+            if (err.code === 'UnauthorizedException') {
+                this.ssoAccessTokenProvider.invalidate()
+            }
+            vscode.window.showErrorMessage(
+                'Something went wrong loading your SSO credentials, please re-initiate login.'
```

Comment:
Localize user-facing message

Reply:
- never need to use "please" in messages
- avoid colloquial language ("Something went wrong")

```suggestion
                'Failed to load SSO credentials. Try logging in again.'
```

Reply:
Added to package.nls, using localized text and command, and using the suggested phrasing.